prov/shm/src/smr_ep.c

```diff
@@ -290,23 +296,23 @@ void smr_progress_cmd(struct smr_ep *ep)
 		}
 		entry = container_of(dlist_entry, struct smr_ep_entry, entry);
 
-		switch (cmd->hdr.op.op_data){
+		switch (cmd->hdr.op.op_src){
 		case smr_src_inline:
 			ret = smr_progress_inline(cmd, entry);
 			break;
 		case smr_src_inject:
 			ret = smr_progress_inject(cmd, entry, ep);
 			break;
 		case smr_src_iov:
-			ret = smr_progress_iov(cmd, entry, ep, cmd->hdr.op.flags);
+			ret = smr_progress_iov(cmd, entry, ep);
 			break;
 		default:
 			FI_WARN(&smr_prov, FI_LOG_EP_CTRL,
 				"unidentified operation type\n");
-			ret = -FI_EINVAL;
+			entry->resv = FI_EINVAL;
 		}
-		ep->rx_comp(ep, entry->context, 0, ret, NULL, &addr,
-			    cmd->hdr.op.tag);
+		ep->rx_comp(ep, entry->context, FI_RECV | entry->flags,
```

Comment:
You may be able to set FI_RECV at the lowest layer -- i.e. wherever rx_comp() points at.